imap/http_caldav.c

```diff
@@ -1528,22 +1528,25 @@ static int caldav_delete_cal(struct transaction_t *txn,
         /* XXX - after legacy records are gone, we can strip this and just not send a
          * cancellation if deleting a record which was never replied to... */
 
-        char *userid = mboxname_to_userid(txn->req_tgt.mbentry->name);
+        char *cal_ownerid = mboxname_to_userid(txn->req_tgt.mbentry->name);
+        char *sched_userid = (txn->req_tgt.flags == TGT_DAV_SHARED) ?
```

Comment:
I suggest to not allow sched_userid being NULL, even if that makes handling the allocated strings here a bit more complicated.

First, it was not clear to me when I saw this PR if sched_userid is nullable and most of the changes in this PR do not pass NULL, even if both cal_ownerid and sched_userid point to the same memory.

Second, I prefer we force us to always think about the schedule id when we call the scheduling layer. JMAP Calendars team mode calendars will require us to do anyway, so it's better to be explicit about it at all call sites.

Reply:
Should be fixed